internal/controller/inference_pool.go

```diff
@@ -0,0 +1,409 @@
+	}
+}
+
+// Reconcile implements the [reconcile.TypedReconciler] for [gwaiev1a2.InferencePool].
+func (c *InferencePoolController) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	var inferencePool gwaiev1a2.InferencePool
+	if err := c.client.Get(ctx, req.NamespacedName, &inferencePool); err != nil {
+		if client.IgnoreNotFound(err) == nil {
+			c.logger.Info("Deleting InferencePool",
+				"namespace", req.Namespace, "name", req.Name)
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	c.logger.Info("Reconciling InferencePool", "namespace", req.Namespace, "name", req.Name)
+	if err := c.syncInferencePool(ctx, &inferencePool); err != nil {
+		c.logger.Error(err, "failed to sync InferencePool")
+		c.updateInferencePoolStatus(ctx, &inferencePool, "NotAccepted", err.Error())
+		return ctrl.Result{}, err
+	}
+	c.updateInferencePoolStatus(ctx, &inferencePool, "Accepted", "InferencePool reconciled successfully")
+	return ctrl.Result{}, nil
+}
+
+// syncInferencePool is the main logic for reconciling the InferencePool resource.
+// This is decoupled from the Reconcile method to centralize the error handling and status updates.
+func (c *InferencePoolController) syncInferencePool(ctx context.Context, inferencePool *gwaiev1a2.InferencePool) error {
+	// Check if the ExtensionReference service exists.
+	if err := c.validateExtensionReference(ctx, inferencePool); err != nil {
+		return err
+	}
+
+	referencedGateways, err := c.getReferencedGateways(ctx, inferencePool)
+	if err != nil {
+		return err
+	}
+
+	c.logger.Info("Found referenced Gateways", "count", len(referencedGateways), "inferencePool", inferencePool.Name)
+	return nil
+}
+
+// routeReferencesInferencePool checks if an AIGatewayRoute references the given InferencePool.
+func (c *InferencePoolController) routeReferencesInferencePool(route *aigv1a1.AIGatewayRoute, inferencePoolName string) bool {
+	for _, rule := range route.Spec.Rules {
+		for _, backendRef := range rule.BackendRefs {
+			if backendRef.IsInferencePool() && backendRef.Name == inferencePoolName {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// getReferencedGateways returns all Gateways that reference the given InferencePool.
+func (c *InferencePoolController) getReferencedGateways(ctx context.Context, inferencePool *gwaiev1a2.InferencePool) (map[string]*gwapiv1.Gateway, error) {
+	// Find all Gateways across all namespaces.
+	var gateways gwapiv1.GatewayList
+	if err := c.client.List(ctx, &gateways); err != nil {
+		return nil, fmt.Errorf("failed to list Gateways: %w", err)
+	}
+
+	referencedGateways := make(map[string]*gwapiv1.Gateway)
+
+	// Check each Gateway to see if it references this InferencePool through routes.
+	for _, gw := range gateways.Items {
+		if c.gatewayReferencesInferencePool(ctx, &gw, inferencePool.Name, inferencePool.Namespace) {
+			gatewayKey := fmt.Sprintf("%s/%s", gw.Namespace, gw.Name)
+			referencedGateways[gatewayKey] = &gw
+		}
+	}
```

Comment:
same comment about escaping
```suggestion
	for i := range gateways.Items {
        gw := &gateways.Items[i]
		if c.gatewayReferencesInferencePool(ctx, gw, inferencePool.Name, inferencePool.Namespace) {
			gatewayKey := fmt.Sprintf("%s/%s", gw.Namespace, gw.Name)
			referencedGateways[gatewayKey] = gw
		}
	}
```